### PR TITLE
feature: Add JSON Lines serializer

### PR DESCRIPTION
**Pull request by bveeramani**
*Issue #, if available:*

*Description of changes:*
* Added `JSONLinesSerializer` to `serializers.py`

*Testing done:*
* Added appropriate tests to `test_serializers.py`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/serializers.py [hunk 1]
```diff
@@ -192,6 +192,34 @@ def serialize(self, data):
...
+    """Serialize data to a JSON Lines formatted string."""
+
+    CONTENT_TYPE = "application/jsonlines"
+
+    def serialize(self, data):
+        """Serialize data of various formats to a JSON Lines formatted string.
+
+        Args:
+            data (object): Data to be serialized. The data can be a string,
+                list of JSON serializable objects, or a file-like object.
```
Comment (metrizable) [inline src/sagemaker/serializers.py:205]: Do you want to support any `Iterable` of JSON serializable objects?

File: src/sagemaker/serializers.py [hunk 2]
```diff
@@ -192,6 +192,34 @@ def serialize(self, data):
...
+
+        Returns:
+            str: The data serialized as a string containing newline-separated
+                JSON values.
+        """
+        if isinstance(data, str):
+            return data
+
+        if isinstance(data, list):
```
Comment (metrizable) [inline src/sagemaker/serializers.py:214]: Should the serializer support the more generic `Iterable` from `collections.abc`?

```suggestion
        if isinstance(data, Iterable):
```

File: src/sagemaker/serializers.py [hunk 3]
```diff
@@ -192,6 +192,34 @@ def serialize(self, data):
...
+            return "\n".join(json.dumps(element) for element in data)
```
Comment (metrizable) [inline src/sagemaker/serializers.py:215]: This code will not serialize the last line correctly, according to the POSIX standard:

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

File: src/sagemaker/serializers.py (hunk 1 above)
Comment (bveeramani) [reply src/sagemaker/serializers.py]: Fixed

File: src/sagemaker/serializers.py (hunk 2 above)
Comment (bveeramani) [reply src/sagemaker/serializers.py]: Fixed

**Pull request merged**